utils/R/convert.input.R

```diff
@@ -5,228 +5,245 @@
-  # Perform Conversion 
-  conversion = "local.remote" #default
+  # Perform Conversion
+  conversion <- "local.remote"  #default
   
-  if(!is.null(browndog) && host$name == 'localhost'){ 
+  if (!is.null(browndog) && host$name == "localhost") {
     # perform conversions with Brown Dog - only works locally right now
-    library(RCurl) 
+    library(RCurl)
     
-    # Determine outputtype using formatname and mimetype of output file
-    # Add issue to github that extension of formats table to include outputtype 
-    if(mimetype ==  'application/x-netcdf') { # Convert to netcdf - only using localhost
-      outputtype <- 'pecan.zip'
-    } else { # Convert to model specific format
-      if(formatname == 'ed.met_driver_header_files_format' ||
-         formatname == 'ed.met_driver_header files format') {
-        outputtype <- 'ed.zip'
-      } else if(formatname == 'Sipnet.climna') {
-        outputtype <- 'clim'
-      } else if(formatname == 'DALEC meteorology') {
-        outputtype <- 'dalec.dat'
-      } else if(formatname == 'LINKAGES met') {
-        outputtype <- 'linkages.dat'
+    # Determine outputtype using formatname and mimetype of output file Add issue to
+    # github that extension of formats table to include outputtype Convert to netcdf
+    # - only using localhost
+    if (mimetype == "application/x-netcdf") {
+      outputtype <- "pecan.zip"
+    } else {
+      # Convert to model specific format
+      if (formatname == "ed.met_driver_header_files_format" || formatname == 
+          "ed.met_driver_header files format") {
+        outputtype <- "ed.zip"
+      } else if (formatname == "Sipnet.climna") {
+        outputtype <- "clim"
+      } else if (formatname == "DALEC meteorology") {
+        outputtype <- "dalec.dat"
+      } else if (formatname == "LINKAGES met") {
+        outputtype <- "linkages.dat"
+      } else {
+        logger.severe(paste("Unknown formatname", formatname))
       }
     }
     
     # create curl options
     if (!is.null(browndog$username) && !is.null(browndog$password)) {
-      userpwd <- paste(browndog$username, browndog$password, sep=":")
+      userpwd <- paste(browndog$username, browndog$password, sep = ":")
       curloptions <- list(userpwd = userpwd, httpauth = 1L)
     }
-    curloptions <- c(curloptions, followlocation=TRUE)
+    curloptions <- c(curloptions, followlocation = TRUE)
     
-    # check if we can do conversion 
-    out.html <- getURL(paste0(
-      "http://dap-dev.ncsa.illinois.edu:8184/inputs/", browndog$inputtype), .opts = curloptions)
-    if(outputtype %in% unlist(strsplit(out.html, '\n'))){
-      logger.info(paste("Conversion from", browndog$inputtype, "to", outputtype, "through Brown Dog"))
+    # check if we can do conversion
+    out.html <- getURL(paste0("http://dap-dev.ncsa.illinois.edu:8184/inputs/", 
+                              browndog$inputtype), .opts = curloptions)
```

Comment:
@robkooper should the URL be hard coded or a variable?  Seems like we shouldn't be hard coded to hit the DAP dev machine.